File: LeetCode/0053_Maximum_Subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]
        if len(nums) == 2:
            return max(nums[0] + nums[1], max(nums[0], nums[1]))
        
        current_sum = 0
        max_sum = nums[0]
        
        for num in nums:
            if current_sum < 0:
                current_sum = 0
            
            current_sum += num
            max_sum = max(current_sum, max_sum)
            
        return max_sum

File: LeetCode/Java/0053_Maximum_Subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        if (nums.length == 1)
            return nums[0];
        if (nums.length == 2)
            return Math.max(nums[0] + nums[1], Math.max(nums[0], nums[1]));
        
        int currentSum = 0;
        int maxSum = nums[0];
        
        for (int num: nums) {
            if (currentSum < 0)
                currentSum = 0;
            currentSum += num;
            maxSum = Math.max(currentSum, maxSum);
        }
        
        return maxSum;
    }
}
